keel-sql/src/main/kotlin/com/netflix/spinnaker/keel/sql/ArtifactUtils.kt

```diff
@@ -189,15 +189,16 @@ internal fun ArtifactVersionSelectStep.fetchSortedArtifactVersions(
  */
 private fun filterDockerVersions(artifact: DockerArtifact, versions: List<PublishedArtifact>): List<PublishedArtifact> =
   versions
-    .filter { shouldInclude(it.version, artifact) }
+    .filter { filterDockerTag(it.version, artifact) }
 
 /**
- * Returns true if a docker tag is not a match to the regex produces exactly one capture group on the tag, false otherwise.
+ * @return true if a docker tag is a match to the regex associated with the [TagVersionStrategy] and produces exactly
+ * one capture group on the tag, false otherwise. This only applies if the artifact is defined in terms of versions.
  */
-internal fun shouldInclude(tag: String, artifact: DockerArtifact) =
-  if (artifact.tagVersionStrategy == null) {
-    log.warn("Attempt to check Docker tag against unspecified tag version strategy for $artifact. Excluding.")
-    false
+internal fun filterDockerTag(tag: String, artifact: DockerArtifact) =
+  if (artifact.filteredBySource) {
```

Comment:
This was the bug. Instead of checking if the tag version strategy is `null`, I thought it was clearer to check if the artifact is filtered by source, which means the filter doesn't apply, and so we return `true`.